### PR TITLE
Extract common interface to a single mixin.

### PR DESCRIPTION
**Pull request by terrellt**

### CODE REVIEW

**Comment by terrellt** [comment]
@elrayle @jcoyne Can you take a look?

**Comment by mjgiarlo** [comment]
:+1:

File: spec/hydra/pcdm/collection_indexer_spec.rb [hunk 1]
```diff
@@ -9,7 +9,9 @@
 
   before do
     allow(collection).to receive(:child_collections).and_return([child_collections1, child_collections2])
```
Comment (jcoyne) [inline spec/hydra/pcdm/collection_indexer_spec.rb]: Do we still need this now that we are indexing child_collection_ids?

File: spec/hydra/pcdm/collection_indexer_spec.rb [hunk 2]
```diff
@@ -9,7 +9,9 @@
...
+    allow(collection).to receive(:child_collection_ids).and_return(collection.child_collections.map(&:id))
     allow(collection).to receive(:child_objects).and_return([child_object1])
```
Comment (jcoyne) [inline spec/hydra/pcdm/collection_indexer_spec.rb]: Do we still need this now that we are indexing child_object_ids?

**Comment by terrellt** [comment]
@jcoyne Cleaned up that spec.

**Pull request merged**